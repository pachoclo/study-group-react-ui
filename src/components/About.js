import React from 'react'

const About = ({ auth }) => (
  <>
    <h2>About</h2>
    {auth.isAuthenticated() && <p>(c) Study Groupssss weee</p>}
    {!auth.isAuthenticated() && <p>Not logged In!!! Please login mr/mrs...</p>}
  </>
)

export default About
