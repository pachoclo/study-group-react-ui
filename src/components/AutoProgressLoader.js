import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'

const defaults = {
  initialProgress: 5,
  intervalMillis: 500,
  increment: 5
}

/**
 * Renders a loader that automatically increments its progress
 *
 * @param {number} intervalMillis specifies how often the progress should be increased
 */
export const AutoProgressLoader = ({ intervalMillis = defaults.intervalMillis }) => {
  const [progress, setProgress] = useState(defaults.initialProgress)

  useEffect(() => {
    const intervalId = setInterval(() => {
      setProgress(currentProgress => {
        const newProgress = currentProgress + 5
        return newProgress <= 100 ? newProgress : currentProgress
      })
    }, intervalMillis)

    return function cleanUp () {
      clearInterval(intervalId)
      setProgress(100)
    }
  }, [intervalMillis])

  return (
    <div className='progress' style={{ height: '4px', backgroundColor: 'gray', borderRadius: 0 }}>
      <div
        className='progress-bar progress-bar-striped progress-bar-animated bg-warning'
        role='progressbar'
        aria-valuenow={progress}
        aria-valuemin='0'
        aria-valuemax='100'
        style={{ width: `${progress}%` }}
      />
    </div>
  )
}

AutoProgressLoader.propTypes = {
  intervalMillis: PropTypes.number
}

export default AutoProgressLoader
