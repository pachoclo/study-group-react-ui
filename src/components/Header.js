import React from 'react'
import Menu from './Menu'
import { withRouter } from 'react-router-dom'

const Header = props => <Menu {...props} />

export default withRouter(Header)
