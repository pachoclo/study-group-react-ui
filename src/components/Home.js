import React from 'react'
import { connect } from 'react-redux'
import {} from '../store/actions/authActions'

const Home = ({ isAuthenticated, profile }) => (
  <>
    <h2>Home</h2>
    {isAuthenticated && (
      <>
        <p>La casaa</p>
        <p>Profile:</p>
        <pre> {JSON.stringify(profile, null, 2)}</pre>
      </>
    )}
    {!isAuthenticated && <p>Not logged In!!! Please login mr/mrs...</p>}
  </>
)

const mapStateToProps = ({ auth }) => ({
  isAuthenticated: auth.authenticated,
  profile: auth.profile
})

export default connect(mapStateToProps)(Home)
