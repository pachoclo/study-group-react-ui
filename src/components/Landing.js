import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { preventDefault } from '../util'
import { loginRequest } from '../redux/actions/authActions'

const Landing = ({ isAuthenticated, loginRequest, loading }) => {
  return (
    <div className='landing'>
      <div className='dark-overlay landing-inner text-light'>
        <div className='container'>
          <div className='row'>
            <div className='col-md-12 text-center'>
              <h1 className='display-3 mb-4'>StudyGroup</h1>
              <p className='lead'>
                Gets you working with your peers pronto.
                <br />
                Setup meetings and organize your materials in one place
              </p>
              {!isAuthenticated && !loading && (
                <a href='login.html' className='btn btn-lg btn-info mt-5' onClick={preventDefault(loginRequest)}>
                  Login
                </a>
              )}
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

const mapStateToProps = ({ auth, loader }) => ({
  isAuthenticated: auth.authenticated,
  loading: loader.loading
})

Landing.propTypes = {
  isAuthenticated: PropTypes.bool,
  loginRequest: PropTypes.func,
  loading: PropTypes.bool
}

export default connect(
  mapStateToProps,
  { loginRequest }
)(Landing)
