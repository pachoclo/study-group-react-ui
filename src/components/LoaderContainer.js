import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import AutoProgressLoader from './AutoProgressLoader'

export const LoaderContainer = ({ loading, intervalMillis }) => {
  return loading ? <AutoProgressLoader intervalMillis={intervalMillis} /> : null
}

LoaderContainer.propTypes = {
  loading: PropTypes.bool,
  intervalMillis: PropTypes.number
}

const mapStateToProps = ({ loader }) => loader

export default connect(mapStateToProps)(LoaderContainer)
