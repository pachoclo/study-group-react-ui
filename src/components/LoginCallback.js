import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Redirect } from 'react-router-dom'
import { loginResponse } from '../redux/actions/authActions'
import Landing from './Landing'

export const LoginCallback = ({ isAuthenticated, handleLoginResponse, location }) => {
  if (!isAuthenticated && /access_token|id_token|error/.test(location.hash)) {
    handleLoginResponse()
    return <Landing />
  }
  return <Redirect to={'/'} push />
}

LoginCallback.propTypes = {
  isAuthenticated: PropTypes.bool,
  handleLoginResponse: PropTypes.func,
  location: PropTypes.object
}

const mapStateToProps = ({ auth }) => ({
  isAuthenticated: auth.authenticated
})

const actions = {
  handleLoginResponse: loginResponse
}

export default connect(
  mapStateToProps,
  actions
)(LoginCallback)
