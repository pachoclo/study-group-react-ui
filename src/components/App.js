import React, { useEffect } from 'react'
import { BrowserRouter as Router, Route } from 'react-router-dom'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { renewSession } from '../redux/actions/authActions'
import '../App.css'
import authClient from '../Auth/Auth'
import LoginCallback from './LoginCallback'
import Navbar from './layout/Navbar'
import ErrorAlert from './layout/ErrorAlert'
import Landing from './layout/Landing'
import TopLevelLoader from './TopLevelLoader'

function App ({ renewSession, error }) {
  useEffect(() => {
    // if the user hasn't logged out, try to renew the session
    authClient.isLoggedIn() && renewSession()
  })

  return (
    <Router>
      <>
        <Navbar />
        <TopLevelLoader />
        <div className='{//container}'>
          {error.err && <ErrorAlert error={error.err} followUp={error.followUp} />}
          <Route exact path='/' component={Landing} />
          <Route path='/loginCallback' component={LoginCallback} />
        </div>
      </>
    </Router>
  )
}

App.propTypes = {
  renewSession: PropTypes.func,
  error: PropTypes.object
}

const mapStateToProps = ({ error }) => ({ error })

export default connect(
  mapStateToProps,
  { renewSession }
)(App)
