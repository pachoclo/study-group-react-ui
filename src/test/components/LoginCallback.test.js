import React from 'react'
import { LoginCallback } from '../../components/LoginCallback'
import { shallow } from 'enzyme'

describe('LoginCallback Test', () => {
  it('should render correctly with default state', () => {
    const handleLoginResponseMock = jest.fn()
    const props = {
      isAuthenticated: false,
      location: {
        hash: 'access_token'
      },
      handleLoginResponse: handleLoginResponseMock
    }
    const component = shallow(<LoginCallback {...props} />)
    expect(handleLoginResponseMock).toHaveBeenCalled()
    expect(component).toMatchSnapshot()
  })

  it('should redirect if url does not contain the token', () => {
    const handleLoginResponseMock = jest.fn()
    const regexTestSpy = jest.spyOn(RegExp.prototype, 'test')
    const props = {
      isAuthenticated: false,
      location: {
        hash: null
      },
      handleLoginResponse: handleLoginResponseMock
    }
    const component = shallow(<LoginCallback {...props} />)
    expect(handleLoginResponseMock).not.toHaveBeenCalled()
    expect(regexTestSpy).toHaveBeenCalled()
    expect(component).toMatchSnapshot()
  })

  it('should render a redirect if the user is authenticated', () => {
    const handleLoginResponseMock = jest.fn()
    const props = {
      isAuthenticated: true,
      location: { hash: null },
      handleLoginResponse: handleLoginResponseMock
    }
    const component = shallow(<LoginCallback {...props} />)
    expect(handleLoginResponseMock).not.toHaveBeenCalled()
    expect(component).toMatchSnapshot()
  })
})
