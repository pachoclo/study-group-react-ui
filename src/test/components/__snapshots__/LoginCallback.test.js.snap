// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginCallback Test should redirect if url does not contain the token 1`] = `
<Redirect
  push={true}
  to="/"
/>
`;

exports[`LoginCallback Test should render a redirect if the user is authenticated 1`] = `
<Redirect
  push={true}
  to="/"
/>
`;

exports[`LoginCallback Test should render correctly with default state 1`] = `<Connect(Landing) />`;
